fix(CSS): add a note to table-layout `fixed` value section

OnkarRuikar opened the pull request:
- Fixes: https://github.com/mdn/content/issues/27728

OnkarRuikar committed:
fix(CSS): add a note to table-layout `fixed` value section

dipikabh reviewed (changes requested):
Thanks, @OnkarRuikar.

I am not sure if this note text will resolve the reported issue. Using the example on the page, with `table-layout: fixed;` if I remove `width: 120px;`, then adding more content in any row increases the width of the column.

I think the current sentence "Cells in subsequent rows do not affect column widths." is getting read as "_Content in_ cells in subsequent rows do not affect column widths.", when it probably means "_Width of_ cells in subsequent rows do not affect column widths." (the first sentence says "Table and column widths are set by the widths of table and col elements or by the _width of the first row of cells_.")

How about we remove "Cells in subsequent rows do not affect column widths." to remove the confusion. And instead move this sentence from the next para "Cells use the [overflow](https://developer.mozilla.org/en-US/docs/Web/CSS/overflow) property to determine whether to clip any overflowing content, but only if the table has a known width; otherwise, they won't overflow the cells." in its place?

Also we might want to begin the description for the `fixed` keyword with:
"Uses the fixed table layout algorithm, in which the layout of the table depends on the table’s width, the width of the columns, and borders or cell spacing, not on the content of the cells." (probably use this to replace the first sentence in `fixed`).

Let me know what you think.

OnkarRuikar commented:
> Using the example on the page, with `table-layout: fixed;` if I remove `width: 120px;`, then adding more content in any row increases the width of the column.

:D By removing the `width` you are making it `auto`; it no longer remained the fixed layout algorithm!  With no fixed width specified, the table has to rely on its contents to get the size. Hence `table-layout: fixed` is ignored and the usual auto layout algorithm is used.

> I think the current sentence "Cells in subsequent rows do not affect column widths." is getting read as "Content in cells in subsequent rows do not affect column widths.", when it probably means "Width of cells in subsequent rows do not affect column widths." (the first sentence says "Table and column widths are set by the widths of table and col elements or by the width of the first row of cells.")

:woozy_face: I think summarizing the fixed width algorithm in few sentences is not helping here. Let's provide the algorithm itself.

OnkarRuikar committed:
provide the entire algorithm

OnkarRuikar committed:
fix prettier issues

dipikabh commented on files/en-us/web/css/table-layout/index.md:
```diff
@@ -29,12 +29,21 @@ table-layout: unset;
 - `fixed`
 
-  - : Table and column widths are set by the widths of `table` and `col` elements or by the width of the first row of cells. Cells in subsequent rows do not affect column widths.
+  - : For this value the table's _width needs to be specified explicitly_ with the `width` property. The `width` value `auto` makes UA use the automatic table layout algorithm, in such case this value has no effect.\
+    This is a faster algorithm, because the horizontal layout of the table depends only on the table's width, the width of the columns, and borders or cell spacing. The horizontal layout doesn't depend on the contents of the cells, because it depends only on explicitly set widths.
+
+    In the fixed table layout algorithm, the width of each column is determined as follows:
+
+    - A column element with explicit width sets the width for that column.
+    - Otherwise, a cell in the first row with explicit width determines the width for that column.
+    - Otherwise, the column gets the width from the shared remaining horizontal space.
```
- added some edits for parallel construction
- other edits to slightly improve the clarity
- removed commas before 'because'

```suggestion
  - : The fixed table layout algorithm is used. When using this keyword, the table's width _needs to be specified explicitly_ using the [`width`](/en-US/docs/Web/CSS/width) property. If the value of the `width` property is set to `auto`or is not specified, the browser uses the automatic table layout algorithm, in which case the `fixed` value has no effect.\
    The fixed table layout algorithm is faster than the automatic layout algorithm because the horizontal layout of the table depends only on the table's width, the width of the columns, and borders or cell spacing. The horizontal layout doesn't depend on the contents of the cells because it depends only on explicitly set widths.

    In the fixed table layout algorithm, the width of each column is determined as follows:

    - A column element with explicit width sets the width for that column.
    - Otherwise, a cell in the first row with explicit width determines the width for that column.
    - Otherwise, the column gets the width from the shared remaining horizontal space.
```

dipikabh reviewed (changes requested):
Thanks @OnkarRuikar for adding these updates. I've suggested some edits.

> By removing the width you are making it auto; it no longer remained the fixed layout algorithm

I was trying to replicate the issue posted by the OP :) And so I was suggesting a reword of the `fixed` keyword description

OnkarRuikar committed:
Apply suggestions from code review

Co-authored-by: Dipika Bhattacharya <dipika@foss-community.org>

dipikabh reviewed (approved):
Thank you 👍
